### ISSUE TITLE
vector_set_uncertainty tries to index uncertainty, even if it's None

### ISSUE THREAD

**Issue by wmwv** [Autostronomy/AstroPhot#133]
**Describe the bug**

`vector_set_uncertainty` still indexes `_uncertainty` with `mask` without checking to see if `_uncertainty` is None.  But if` _uncertainty` is None, this fails with:

```
    def vector_set_uncertainty(self, uncertainty):
        """Update the uncertainty vector for this parameter DAG (see
        vector_set_values).

        """
        uncertainty = torch.as_tensor(uncertainty, dtype = AP_config.ap_dtype, device = AP_config.ap_device)
        if self.leaf:
>           self._uncertainty[self.mask] = uncertainty
E           TypeError: 'NoneType' object does not support item assignment
```

**To Reproduce**
```
P1 = Parameter_Node("test_vector_set_uncertainty_that_was_none", value=[0, 0], uncertainty=None)
P1.vector_set_uncertainty([1, 1])
```

I've added a test  that triggers this in `tests/test_parameter.py` under a new `test_parameter_uncertainty.py` function.

**Expected behavior**
I expected that vector_set_uncertainty would take the `None` value an updated it to [1, 1].

**Suggested solutions**
1. vector_set_uncertainty should check if there is a shape mismatch.  I don't fully understand how to do this through the DAG hierarchy.
2. Consider enforcing that the shape of `uncertainty` matches that of `value` on intialization.  I think this could be done by re-writing the `__init__` of the parameter node class to use the class setters to construct the initial values.  I'm not sure I know that `uncertainty` should be if `value` is a function.

I would suggest doing both of these.

**Comment by ConnorStoneAstro** [comment]
Hi @wmwv This was partially fixed in the last update. When one sets `param.uncertainty` it will ensure that the shape matches `value`. Looks like we should similarly fix the `vector_set_uncertainty` function. I will add this fix to PR #132 as it is somewhat related

**Comment by ConnorStoneAstro** [comment]
I think if `value` is a function then I should use basic error propagation if someone calls the uncertainty function. Internally though it would never get used since function nodes are not leaf nodes so they don't really play into the vector quantities.

**Comment by ConnorStoneAstro** [comment]
> I think if `value` is a function then I should use basic error propagation if someone calls the uncertainty function. Internally though it would never get used since function nodes are not leaf nodes so they don't really play into the vector quantities.

That can be for a future PR though, it's not really needed right away.

**Comment by ConnorStoneAstro** [comment]
This is closed by #132
